clientgui/BOINCListCtrl.cpp

```diff
@@ -240,42 +239,67 @@ bool CBOINCListCtrl::OnRestoreState(wxConfigBase* pConfig) {
         //
         // This will also be triggered if the locale is changed, which will cause 
         // SetListColumnOrder() to be called again so the wxListCtrl will be set 
-        // up with the correctly labeled columns. 
+        // up with the correctly labeled columns.
+        //
         bool foundNewColumns = false;
+        bool foundNewDefaultColumns = false;
+        bool foundNewHiddenColumns = false;
         
         if (pConfig->Read(wxT("HiddenColumns"), &strHiddenColumns)) {
             wxArrayString hiddenArray;
+            wxArrayString defaultArray;
             TokenizedStringToArray(strHiddenColumns, ";", &hiddenArray);
             int shownCount = orderArray.size();
             int hiddenCount = hiddenArray.size();
             int totalCount = pView->m_aStdColNameOrder->size();
-            for (int i = 0; i < totalCount; ++i) {
+            int defaultCount = pView->m_iNumDefaultShownColumns;
+            for (int i = 0; i < totalCount; ++i) {  // cycles through updated array of columns.
                 wxString columnNameToFind = pView->m_aStdColNameOrder->Item(i);
                 bool found = false;
-                for (int j = 0; j < shownCount; ++j) {
+                for (int j = 0; j < shownCount; ++j) {  // cycles through list of visible columns.
                     if (orderArray[j].IsSameAs(columnNameToFind)) {
                         found = true;
                         break;
                     }
                 }
                 if (found) continue;
 
-                for (int j = 0; j < hiddenCount; ++j) {
+                for (int j = 0; j < hiddenCount; ++j) {  // cycles through the hidden columns.
                     if (hiddenArray[j].IsSameAs(columnNameToFind)) {
                         found = true;
                         break;
                     }
                 }
                 if (found) continue;
-                
-                foundNewColumns =  true;
-                orderArray.Add(columnNameToFind);
+
+                foundNewColumns = true;
+                // If we got this far, then we know this column is new.
+                // Now it needs to be determined if the new column should be shown by default or not.
+                // Create array of default columns.
+                //
+                defaultArray.Clear();
+                for (int k = 0; k < pView->m_iNumDefaultShownColumns; ++k) {
+                    defaultArray.Add(pView->m_aStdColNameOrder->Item(pView->m_iDefaultShownColumns[k]));
+                }
+                for (int k = 0; k < defaultCount; ++k) {
```

Comment:
On the line 255 you saved `defaultCount` as a value from `pView->m_iNumDefaultShownColumns`. Then you put into `defaultArray` `pView->m_iNumDefaultShownColumns` number of values on the line 281. Here you iterate `defaultArray` using `defaultValue` again. For me it looks not safe and too compicated. Why just don't write
```
for (int k = 0; k < defaultArray.GetCount(); ++k) {
```
instead of 
```
for (int k = 0; k < defaultCount; ++k) {
```
?
Then you don't need `defaultCount` variable at all.

Reply:
I see what you're saying, I agree.  Fixed now.